javaslang-benchmark/src/test/java/javaslang/idiom/ForBenchmark.java

```diff
@@ -0,0 +1,81 @@
+package javaslang.idiom;
+
+import javaslang.JmhRunner;
+import javaslang.collection.Array;
+import javaslang.collection.Iterator;
+import org.junit.Test;
+import org.openjdk.jmh.annotations.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static javaslang.API.For;
+import static javaslang.JmhRunner.getRandomValues;
+
+/**
+ * Benchmark for nested loops vs javaslang's For().yield comprehensions.
+ *
+ * @see javaslang.API.For2
+ */
+public class ForBenchmark {
+    static final Array<Class<?>> CLASSES = Array.of(
+            ForComprehensionNestedFor.class
+    );
+
+    @Test
+    public void testAsserts() {
+        JmhRunner.runDebugWithAsserts(CLASSES);
+    }
+
+    public static void main(String... args) {
+        JmhRunner.runNormalNoAsserts(CLASSES);
+    }
+
+    @State(Scope.Benchmark)
+    public static class Base {
+        @Param({ "10" })
+        public int CONTAINER_SIZE;
+
+        List<Integer> ELEMENTS;
+
+        @Setup
+        public void setup() {
+            Integer[] array = getRandomValues(CONTAINER_SIZE, 0, true);
+            ELEMENTS = Arrays.asList(array);
+        }
+
+        Integer combineNumbers(Integer i, Integer j) {
+            //with random numbers, we need an op that is predictable,
+            //so we XOR both numbers and add them. It will always be zero.
+            //this way we can assert the end result
+            return (i ^ i) + (j ^ j);
```

Comment:
Please stop doing this, `(i ^ i) + (j ^ j)` == 0 and the compiler knows this.
Measure, don't guess!